tools/pipelines-tasks/AVDAppAttachPublish/main.ts

```diff
@@ -0,0 +1,136 @@
+		const segments = pathname.split('/');
+
+		const resourceGroupIndex = segments.indexOf('resourceGroups');
+
+		if (resourceGroupIndex !== -1 && resourceGroupIndex < segments.length - 1) {
+			return segments[resourceGroupIndex + 1];
+		}
+	}
+
+    return "";
+}
+
+async function run(): Promise<void> {
+	let isSuccessful = true;
+	try {
+		await helpers.installNuget(helpers.APPATTACH_FRAMEWORK_NUPKG_DIR, AppAttachFrameworkDll, 'net472');
+
+		let connectedServiceName = tl.getInput('connectedServiceNameARM', true)!;
+
+		let subscriptionID: string = tl.getEndpointDataParameter(connectedServiceName, 'subscriptionid', true)!;
+
+		let storageAccountName = tl.getInput('storageAccount', true)!;
+		var azureEndpoint: AzureEndpoint = await new AzureRMEndpoint(connectedServiceName).getEndpoint();
+		let accessToken: string = (await azureEndpoint.applicationTokenCredentials.getToken()).toString();
+
+		const storageArmClient = new armStorage.StorageManagementClient(azureEndpoint.applicationTokenCredentials, (azureEndpoint.subscriptionID)!);
+		let storageAccount: StorageAccount = await storageArmClient.storageAccounts.get(storageAccountName);
+		let storageAccountResourceGroupName = getResourceGroupNameFromUri(storageAccount.id);
+
+		let accessKeys = await storageArmClient.storageAccounts.listKeys(storageAccountResourceGroupName, storageAccountName, null);
+		let accessKey: string = accessKeys[0];
+		let StorageAccountConnectionString: string = "DefaultEndpointsProtocol=https;AccountName=" + storageAccountName + ";AccountKey=" + accessKey
+			+ ";EndpointSuffix=core.windows.net";
+		let username: string = (process.env.BUILD_REQUESTEDFOREMAIL)!;
+		const PUBLISH_TO_AZURE_TASK = '8';
+
+		const appattachConfig = {
+			'emailId': username,
+			'accessToken': accessToken,
+			'azureStorageKey': StorageAccountConnectionString,
+			'azureWorkspace': tl.getInput('workSpace', true),
+			'azureStorageFileShare': tl.getInput('fileShare', true),
+			'azureStorageAccount': storageAccountName,
+			'azureSubscriptionId': subscriptionID,
+			'azureResourceGroup': tl.getInput('resourceGroupName', true),
+			'azureHostPoolName': tl.getInput('hostPool', true),
+			'azureApplicationGroupName': tl.getInput('applicationGroup', true),
+			'taskType': PUBLISH_TO_AZURE_TASK,
+			'appAttachImagePath': tl.getInput('vhdxPath', true),
+			'clientType': helpers.CLIENT_TYPE,
+			'clientVersion': helpers.CLIENT_VERSION
+		};
+
+		const jsonString = JSON.stringify(appattachConfig);
+		tl.debug(jsonString);
+
+		const powershellRunner: ToolRunner = helpers.getPowershellRunner(HELPER_SCRIPT);
+		powershellRunner.arg(['-inputJsonStr', '\'' + jsonString + '\'']);
+		powershellRunner.arg(['-targetDLL', TARGET_DLL]);
+
+		const execResult = await powershellRunner.execSync();
+
+		if (execResult.code) {
+			throw execResult.stderr;
+		}
+
+		// display log file content to console
+		const appAttachLogDir: string = path.join(os.tmpdir(), 'AppAttach');
```

Comment:
Can we directly push the logs to console, rather reading from file?

Reply:
These logs are extracted from AppAttach Framework logging mechanism which currently supports putting them to a log file.